File: Delgado-Alarcon-Alan_Ignacio/procesos/Python/fork.py
import os
import sys

if len(sys.argv) != 2:
    print("ERR: no params")
    sys.exit(1)

try:
    N = int(sys.argv[1])
except ValueError:
    print("ERR: invalid parameter")
    sys.exit(1)

for i in range(1, N + 1):
    try:
        if os.fork() == -1:
            print("ERROR")
    except OSError:
        print("ERROR")
print(os.getpid())

File: Delgado-Alarcon-Alan_Ignacio/procesos/Rust/fork.rs
use std::env;
use std::process;

fn main() {
    let args: Vec<String> = env::args().collect();
    if args.len() != 2 {
        eprintln!("ERR: no params");
        process::exit(1);
    }

    let n: i32 = match args[1].parse() {
        Ok(num) => num,
        Err(_) => {
            eprintln!("ERR: invalid parameter");
            process::exit(1);
        }
    };

    for _ in 1..=n {
        match unsafe { libc::fork() } {
            -1 => eprintln!("ERROR"),
            _ => {}
        }
    }

    println!("{}", unsafe { libc::getpid() });
}
